reframe/schemas/runreport.json

```diff
@@ -94,10 +95,12 @@
                 "time_run": {"type": ["number", "null"]},
                 "time_sanity": {"type": ["number", "null"]},
                 "time_setup": {"type": ["number", "null"]},
-                "time_total": {"type": ["number", "null"]}
+                "time_total": {"type": ["number", "null"]},
+                "unique_name": {"type": "string"}
             },
             "required": [
-                "environment", "name", "result", "system", "filename"
+                "environment", "fail_phase", "fail_reason", "filename",
+                "result", "stagedir", "system", "time_total", "unique_name"
```

Comment:
Yes, because these are the fields that are actually used either when loading a report or for generating the JUnit report.  Perhaps, those from the JUnit report may not be needed to be honest, since this is always generated fresh from a run.

Reply:
I have now put only those that are strictly necessary for loading a report. The idea is to ensure backward compatibility, so that you are able to load older reports and restore the test cases (although we don't have a schema for the test case dump, but we'll address if we see that this actually becomes a problem).